src/decision/dMCDA.jl

```diff
@@ -689,16 +689,16 @@ function run_site_selection(dom::Domain, scenarios::DataFrame, sum_cover::Abstra
 
     n_sites = length(dom.site_ids)
     for (scen_idx, scen) in enumerate(eachrow(scenarios))
-        depth_criteria = (dom.site_data.depth_med .<= scen.max_depth) .& (dom.site_data.depth_med .>= scen.depth_min)
+        depth_criteria = set_depth_criteria(dom.site_data.depth_med, scen.max_depth, scen.depth_min)
         depth_priority = findall(depth_criteria)
 
         considered_sites = target_site_ids[findall(in(depth_priority), target_site_ids)]
 
         ranks_store(scenarios=scen_idx, sites=dom.site_ids[considered_sites]) .= site_selection(
```

Comment:
I think this code is a little hard to read and in this case it could be useful to assign each function call to a variable and then do the comparison on a separate line..? If you do this with at least `size_selection` it will already be better. Something like the following:

```julia
selected_sites = site_selection(
    dom,
    scen,
    env_mean(wave_scens[:, :, target_wave_scens], (:timesteps, :scenarios)),
    env_mean(dhw_scens[:, :, target_dhw_scens], (:timesteps, :scenarios)),
    considered_sites,
    sum_cover[scen_idx, :],
    area_to_seed
)

ranks_store(scenarios=scen_idx, sites=dom.site_ids[considered_sites]) .= selected_sites
```

Reply:
Can definitely do this

Reply:
Yes, it's more readable, but hurts performance (I expect this approach to use double the memory allocations). MCDA is a performance critical part so I would prefer a comment over readable code in this case.

That said, there are definitely other things here that hurt performance more than the readable code, so I'm fine to concede to @Zapiano 's point here.

Reply:
@ConnectedSystems why does it use double the memory?